### PR TITLE
moved HALO specifics navigation data to function

### PR DESCRIPTION
**Pull request by d70-t**
In order to support multiple platforms, specific code of loading platform navigation data should be factored out to specific functions. These can be collected in navdata.py. Dispatching to the correct function is done based on the `platform` attribute of each flight.

This PR should help with #3 and also includes a method to obtain HALO flight data from Aeris via opendap.

### CODE REVIEW

**Review by RobertPincus — APPROVED** [review]
Hi - This looks great, and I think I can implement soon for P3 too. I'm presuming that, because the catalog doesn't uses authentication for the OpenDAP access, we will be able to use generic versions of  `intake` and `intake-xarray`

File: scripts/navdata.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        _catalog_cache["HALO"] = open_catalog("https://raw.githubusercontent.com/d70-t/eurec4a-intake/HALO_QL/catalog.yml")
+
+    catalog = _catalog_cache["HALO"]
+    bahamas = catalog.halo.bahamas.ql.by_flight_id[flight].to_dask()
+    ds = bahamas.rename({"tid": "time"})
+    return xr.Dataset({
+        "time": ds.TIME,
+        "lat": ds.IRS_LAT,
+        "lon": ds.IRS_LON,
+        "altitude": ds.IRS_ALT,
```
Comment (RobertPincus) [inline scripts/navdata.py]: One quick note: I think the metadata stands specific "alt" instead of "altitude"

Comment (d70-t) [reply scripts/navdata.py]: changed it to "alt"

**Comment by d70-t** [comment]
Thanks @RobertPincus for reviewing, will you merge it if you are happy with the situation?

**Pull request merged**